arch/risc-v/src/common/supervisor/riscv_perform_syscall.c

```diff
@@ -46,6 +46,16 @@ void *riscv_perform_syscall(uintreg_t *regs)
     {
       (*running_task)->xcp.regs = regs;
     }
+  else
```

Comment:
Another option is to set local copy of running_task = NULL if regs[REG_A0] == SYS_restore_context, this would avoid duplicating the addrenv handling code etc.

Reply:
And riscv_doirq does not need patching ? Are we missing call to nxsched_resume_scheduler(*running_task); there ?

Reply:
> Another option is to set local copy of running_task = NULL if regs[REG_A0] == SYS_restore_context, this would avoid duplicating the addrenv handling code etc.

We hope to be able to directly use g_running_task in any scenario without it being NULL.

Reply:
> And riscv_doirq does not need patching ? Are we missing call to nxsched_resume_scheduler(*running_task); there ?

done